src/main/java/com/mongodb/hibernate/jdbc/MongoConnection.java

```diff
@@ -208,15 +210,30 @@ public DatabaseMetaData getMetaData() throws SQLException {
         return null;
     }
 
+    /**
+     * @param schemaName {@inheritDoc}. If {@code null}, then {@link MongoDialectSettings#getDatabaseName()} is used.
+     */
+    @Override
+    public void setSchema(@Nullable String schemaName) throws SQLException {
+        checkClosed();
+        mongoDatabase = mongoDatabase(schemaName);
+    }
+
+    private MongoDatabase mongoDatabase(@Nullable String mongoDatabaseName) {
+        return mongoClient.getDatabase(mongoDatabaseName == null ? config.getDatabaseName() : mongoDatabaseName);
+    }
+
     /**
      * Used during Hibernate's DDL step for Information Extraction purposes.
      *
+     * @return The current schema name.
+     *     <p>Default value: {@link MongoDialectSettings#getDatabaseName()}.
      * @see org.hibernate.tool.schema.extract.internal.AbstractInformationExtractorImpl
      */
     @Override
-    public @Nullable String getSchema() throws SQLException {
+    public String getSchema() throws SQLException {
         checkClosed();
-        return null;
+        return mongoDatabase.getName();
```

Comment:
So the above two methods are only used during Hibernate's DDL step (per our latest design change, we might well delete these two methods for we won't support them at all; DDL is irrelevant to NoSQL; indexing and unique constraint feature later on won't rely on the above methods as well).

Reply:
We agreed not to interpret the MongoDB database as JDBC schema, at least for now.

Done in 3bb8ee965f06d51bdc5b8fd333fa346363017c4b.